lib/mongo/socket.rb

```diff
@@ -334,12 +334,36 @@ def read_from_socket(length, timeout: nil)
             raise Errno::ETIMEDOUT, "Took more than #{_timeout} seconds to receive data"
           end
         end
+        pipe = options[:pipe]
```

Comment:
What happens if pipe is nil here? Will we fail with a meaningful error, or just keep working?

Reply:
wherever a pipe is used we check for its presence. We will keep working.